lib/robots/viomi/capabilities/ViomiZoneCleaningCapability.js

```diff
@@ -0,0 +1,57 @@
+const ZoneCleaningCapability = require("../../../core/capabilities/ZoneCleaningCapability");
+const ViomiMapParser = require("../../../ViomiMapParser");
+const Logger = require("../../../Logger");
+
+class ViomiZoneCleaningCapability extends ZoneCleaningCapability {
+    /**
+     * @param {Array<import("../../../entities/core/ValetudoZone")>} valetudoZones
+     * @returns {Promise<void>}
+     */
+    async start(valetudoZones) {
+        let areas = [];
+
+        valetudoZones.forEach( zone => {
+            const pA = ViomiMapParser.positionToViomi(zone.points.pA.x, zone.points.pA.y);
+            const pB = ViomiMapParser.positionToViomi(zone.points.pB.x, zone.points.pB.y);
+            const pC = ViomiMapParser.positionToViomi(zone.points.pC.x, zone.points.pC.y);
+            const pD = ViomiMapParser.positionToViomi(zone.points.pD.x, zone.points.pD.y);
+
+            for (let j = 0; j < zone.iterations; j++) {
+                areas.push([areas.length,
+                    0,
+                    pA.x.toFixed(4),
+                    pA.y.toFixed(4),
+                    pB.x.toFixed(4),
+                    pB.y.toFixed(4),
+                    pC.x.toFixed(4),
+                    pC.y.toFixed(4),
+                    pD.x.toFixed(4),
+                    pD.y.toFixed(4)
+                ].join("_"));
+            }
+        });
+
+        Logger.trace("areas to clean: ", areas);
+        await this.robot.sendCommand("set_zone", [areas.length].concat(areas), {}).finally(() => {
```

Comment:
Wouldn’t this also need to issue a start command?

See the previous implementation here:
https://github.com/rumpeltux/Valetudo/blob/d5d9ec2fc78a6d290ce0be831a5ffc7c41da8a33/lib/devices/Viomi.js#L589

Reply:
done